src/main/java/bsh/Primitive.java

```diff
@@ -183,6 +183,8 @@ public static Primitive shrinkWrap(Object number) {
         if (!(number instanceof Number))
             throw new InterpreterError("Can only shrink wrap Number types");
         Number value = (Number) number;
+        if (value instanceof Float)
+            return new Primitive(value.floatValue());
```

Comment:
I think this needs to do something similar to int and long, comparing the values instead of the type.

See full function 193 and 195:
https://github.com/beanshell/beanshell/blob/53b0436f4e2777e5ec8c8f60e565c7ef48f3b18b/src/main/java/bsh/Primitive.java#L178-L198

I wonder if this would not be sufficient, to just shrink the double result to float if it fits?

Reply:
If you do the widen float to double, do the operation, and then narrow you will get a different result than doing the operation in float.  Best to keep this the same as Java.

In Java byte and short use int as a carrier type.  Arithmetic on any combination of these types will yield an int.  Best to keep this the same as Java.